robowflex_library/include/robowflex_library/detail/cob4.h

```diff
@@ -0,0 +1,130 @@
+/* Author: Juan D. Hernandez */
+
+#ifndef ROBOWFLEX_COB4_
+#define ROBOWFLEX_COB4_
+
+#include <robowflex_library/planning.h>
+#include <robowflex_library/robot.h>
+
+namespace robowflex
+{
+    /** \cond IGNORE */
+    ROBOWFLEX_CLASS_FORWARD(Cob4Robot);
+    /* \endcond */
+
+    /** \class robowflex::Cob4RobotPtr
+        \brief A shared pointer wrapper for robowflex::Cob4Robot. */
+
+    /** \class robowflex::Cob4RobotConstPtr
+        \brief A const shared pointer wrapper for robowflex::Cob4Robot. */
+
+    /** \brief Convenience class that describes the default setup for Cob4.
+     *  Will first attempt to load configuration and description from the robowflex_resources package.
+     *  See https://github.com/KavrakiLab/robowflex_resources for this package.
+     *  If this package is not available, then fetch_description / fetch_moveit_config packages will be used.
+     */
+    class Cob4Robot : public Robot
+    {
+    public:
+        /** \brief Constructor.
+         */
+        Cob4Robot();
+
+        /** \brief Initialize the robot with arm_left and arm_right kinematics.
+         *  \return True on success, false on failure.
+         */
+        bool initialize();
+
+        /** \brief Inserts the caster links if they don't exist.
+         *  \param[in] doc urdf description to be processed.
+         *  \return True on success.
+         */
+        bool addCastersURDF(tinyxml2::XMLDocument &doc);
```

Comment:
Does this robot also have the exact same casters as the Fetch?

Reply:
You're right. I left this when I used the Fetch specification to implement the one for the Care-O-Bot4. I've cleaned it up.